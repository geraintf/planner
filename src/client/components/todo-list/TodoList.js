import React from 'react';
import PropTypes from 'prop-types';

import List from '../todo/TodoList';
import Todo from '../todo/Todo';

const TodoList = () => {

  const todos = [
    {
      id: 'fwefwef',
      checked: false,
      text: 'text'
    },
    {
      id: 'hrthrthrt',
      checked: false,
      text: 'text'
    },
    {
      id: 'fwefd2d2wef',
      checked: false,
      text: 'text'
    }
  ];

  return (
    <List>
      {
        todos.map((props) => {
          <Todo {...props} />
        })
      }
    </List>

  )

};

TodoList.propTypes = {};

export default TodoList;