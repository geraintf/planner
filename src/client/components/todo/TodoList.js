import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import './TodoList.less';

import Todo from './Todo';
import AddTodo from './AddTodo';

const TodoList = ({ items, toggleTodo, addTodo }) => (
  <Fragment>
    <ul className="todo-list">
      {
        items.map(({ id, ...props }) => (
          <Todo
            key={id}
            {...props}
            onClick={() => toggleTodo(id)}
          />
        ))
      }
    </ul>
    <AddTodo addTodo={addTodo} />
  </Fragment>
);

TodoList.propTypes = {
  addTodo: PropTypes.func.isRequired,
  toggleTodo: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    completed: PropTypes.bool
  })).isRequired
};

export default TodoList;
