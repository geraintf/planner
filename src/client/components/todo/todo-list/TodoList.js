import React from 'react';
import PropTypes from 'prop-types';

import './TodoList.scss';

import TodoItem from '../todo-item/TodoItem';

const TodoList = ({
  items,
  toggleTodo,
  sortable
}) => (
  <ul className="todo-list">
    {
      items.map((itemProps, index) => (
        <TodoItem
          key={itemProps.id}
          {...itemProps}
          onClick={() => toggleTodo(itemProps.id)}
          sortable={sortable}
          index={index}
        />
      ))
    }
  </ul>
);

TodoList.propTypes = {
  toggleTodo: PropTypes.func.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    text: PropTypes.string,
    completed: PropTypes.bool
  })).isRequired,
  sortable: PropTypes.bool
};

TodoList.defaultProps = {
  sortable: false
};

export default TodoList;
