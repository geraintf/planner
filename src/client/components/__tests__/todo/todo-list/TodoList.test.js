import React from 'react';
import { shallow } from 'enzyme';

import TodoList from '../../../todo/todo-list/TodoList';

import mockTodos from '../../../../../../test/mock-data/mock-todos';

describe('TodoList', () => {
  let output;
  let props;

  beforeEach(() => {
    props = {
      items: mockTodos,
      toggleTodo: () => {}
    };
    output = shallow(<TodoList {...props} />);
  });

  it('should render correctly', () => {
    expect(output).toMatchSnapshot();
  });
});
