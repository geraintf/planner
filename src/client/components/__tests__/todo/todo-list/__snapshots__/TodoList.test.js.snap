// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render correctly 1`] = `
<ul
  className="todo-list"
>
  <Connect(TodoItem)
    completed={true}
    id="fewf"
    index={0}
    key="fewf"
    onClick={[Function]}
    sortable={false}
    text="onething"
  />
  <Connect(TodoItem)
    completed={false}
    id="gfgd"
    index={1}
    key="gfgd"
    onClick={[Function]}
    sortable={false}
    text="this is a really long todo so im just adding more words to make it longer, this is a really long todo so im just adding more words to make it longer, this is a really long todo so im just adding more words to make it longer"
  />
  <Connect(TodoItem)
    completed={false}
    id="fewgfdgf"
    index={2}
    key="fewgfdgf"
    onClick={[Function]}
    sortable={false}
    text="onethliling"
  />
</ul>
`;
